### PR TITLE
Don't fail if no engine log

### PR DESCRIPTION
**Pull request by myakove**

### CODE REVIEW

File: log_extractor.py [hunk 1]
```diff
@@ -436,8 +444,45 @@ def parse_logs(self):
...
+
+    job_folder = os.path.join(view_folder, job)
+    if not os.path.exists(path=job_folder):
+        os.mkdir(job_folder)
+
+    build_folder = os.path.join(job_folder, build)
+    if not os.path.exists(path=build_folder):
+        os.mkdir(build_folder)
+
+    job_url = (
```
Comment (cynepco3hahue) [inline log_extractor.py]: hm do not sure if it good enough for example 4.0 job link looks like https://rhev-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/4.0v4-GE-Tier1-x86_64/49/
or if you go via views
https://rhev-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/4.0/view/GE/job/4.0v4-GE-Tier2-x86_64/49/
I believe it better to send link to the build as parameter and if you want to create dedicated directories
you can split by '/' and take relevant part for the build number and for the job name

**Comment by myakove** [comment]
I can get the link with jenkinsapi so all we need is job name and build.

File: log_extractor.py [hunk 2]
```diff
@@ -74,27 +82,30 @@ def _is_host_log(path):
...
         """
         for pattern in self.logs:
             if '/{0}'.format(pattern) in path:
                 return True
         return False
 
     @staticmethod
-    def _generate_host_log_name(path):
+    def _generate_host_log_name(path, dst):
         """
```
Comment (cynepco3hahue) [inline log_extractor.py:100]: I send dst to the __init__, so just change TEST_DST to self.dst

File: log_extractor.py [hunk 3]
```diff
@@ -106,16 +117,15 @@ def _generate_host_log_name(path):
             if dir_name in ['logs', 'hosts-logs']:
                 next_dir = True
         f_new_basename = "{0}_{1}".format(prefix, f_basename)
-        return os.path.join(TEST_DST, f_new_basename)
+        return os.path.join(dst, f_new_basename)
 
-    def extract_all(self, path):
+    def extract_all(self, path, dst):
         """
```
Comment (cynepco3hahue) [inline log_extractor.py:123]: the same comment as above use self.dst

File: log_extractor.py [hunk 4]
```diff
@@ -436,8 +449,52 @@ def parse_logs(self):
...
+    """
+    def get_jenkins_connection():
+        """
+        Get Jenkins connection object
+
+        Returns:
+            Jenkins: Jenkins connection
+        """
+        ssl._create_default_https_context = ssl._create_unverified_context
+        server = "https://rhev-jenkins.rhev-ci-vms.eng.rdu2.redhat.com"
```
Comment (cynepco3hahue) [inline log_extractor.py:473]: Put it under the constants

File: log_extractor.py [hunk 5]
```diff
@@ -436,8 +449,52 @@ def parse_logs(self):
...
+    job_folder = os.path.join(folder, job)
+    if not os.path.exists(path=job_folder):
+        os.mkdir(job_folder)
+
+    build_folder = os.path.join(job_folder, str(build))
+    if not os.path.exists(path=build_folder):
+        os.mkdir(build_folder)
+
+    helper.download_artifact(job_url=job_url, dst=build_folder)
+    log_extractor = LogExtractor(dst=build_folder, logs=TEST_LOGS)
```
Comment (cynepco3hahue) [inline log_extractor.py:492]: I wanted to send it as part of cli arguments --logs=vdsm.log,engine.log,...
What do you think?

**Pull request merged**